assets/js/types/type-defs/payment-method-interface.ts

```diff
@@ -0,0 +1,176 @@
+import type PaymentMethodIcons from '@woocommerce/base-components/cart-checkout/payment-method-icons';
+import type LoadingMask from '@woocommerce/base-components/loading-mask';
+
+/**
+ * Internal dependencies
+ */
+import type { Currency } from './currency';
+import type {
+	CartBillingAddress,
+	CartShippingPackageShippingRate,
+} from './cart';
+import type {
+	responseTypes,
+	noticeContexts,
+} from '../../base/context/hooks/use-emit-response';
+import type { emitterCallback } from '../../base/context/event-emit';
+import type { PaymentMethodCurrentStatusType } from '../../base/context/providers/cart-checkout/payment-methods/types';
+import type {
+	CartResponseShippingAddress,
+	CartResponseCouponItem,
+	CartResponseItem,
+	CartResponseFeeItem,
+	ExtensionsData,
+} from './cart-response';
+import type { SHIPPING_ERROR_TYPES } from './contexts';
+
+export interface PreparedCartTotalItem {
+	// The label for the total item.
+	label: string;
+	// The value for the total item.
+	value: number;
+}
+
+export interface BillingDataProps {
+	// All the coupons that were applied to the cart/order.
+	appliedCoupons: CartResponseCouponItem[];
+	//The address used for billing.
+	billingData: CartBillingAddress;
+	//The total item for the cart.
+	cartTotal: PreparedCartTotalItem;
+	// The various subtotal amounts.
+	cartTotalItems: PreparedCartTotalItem[];
+	// Currency object.
+	currency: Currency;
+	// The customer Id the order belongs to.
+	customerId: number;
+	//  True means that the site is configured to display prices including tax.
+	displayPricesIncludingTax: boolean;
+}
+
+export interface CartDataProps {
+	cartItems: CartResponseItem[];
+	cartFees: CartResponseFeeItem[];
+	extensions: ExtensionsData;
+}
+
+export interface CheckoutStatusProps {
+	// If true then totals are being calculated in the checkout.
+	isCalculating: boolean;
+	// If true then the checkout has completed it's processing.
+	isComplete: boolean;
+	// If true then the checkout is idle (no  activity happening).
+	isIdle: boolean;
+	// If true then checkout is processing (finalizing) the order with the server.
+	isProcessing: boolean;
+}
+
+export interface ComponentProps {
+	// A wrapper component used for showing a loading state when the isLoading prop is true.
+	LoadingMask: typeof LoadingMask;
+	// A component used for displaying payment method icons.
+	PaymentMethodIcons: typeof PaymentMethodIcons;
+	// A component used for displaying payment method labels, including an icon.
+	PaymentMethodLabel: typeof PaymentMethodLabel;
+	// A container for holding validation errors
+	ValidationInputError: () => JSX.Element | null;
+}
+
+export interface EmitResponseProps {
+	// Response types that can be returned from emitter observers.
+	responseTypes: responseTypes;
+	// Available contexts that can be returned as the value for the messageContext property on the object  returned from an emitter observer.
+	NoticeContexts: noticeContexts;
```

Comment:
this should be `noticeContexts` instead of `NoticeContexts`. Changing this will result in an TS warning for `noticeContexts` in usePaymentMethodInterface.